pkg/reconciler/build/build.go

```diff
@@ -152,11 +170,45 @@ func (c *Reconciler) reconcile(ctx context.Context, build *buildapi.Build) error
 
 	build.Status.PodName = pod.Name
 	build.Status.StepStates = stepStates(pod)
-	build.Status.StepsCompleted = stepCompleted(pod)
-	build.Status.Conditions = conditionForPod(pod)
+	build.Status.StepsCompleted = stepsCompleted(pod)
+	build.Status.Conditions = conditionForPod(pod, build.Status.StepsCompleted)
 	return nil
 }
 
+func (c *Reconciler) setBuildReady(ctx context.Context, pod *corev1.Pod) (*corev1.Pod, error) {
+	if _, found := pod.Annotations[buildapi.BuildReadyAnnotation]; found {
+		return pod, nil
+	}
+
+	for _, container := range pod.Status.ContainerStatuses {
+		if !container.Ready {
+			return pod, nil
+		}
+	}
+
+	patch, err := json.Marshal([]jsonpatch.JsonPatchOperation{{
+		Operation: "replace",
+		Path:      "/metadata/annotations/" + strings.Replace(buildapi.BuildReadyAnnotation, "/", "~1", 1),
+		Value:     "true",
+	}})
```

Comment:
Does the replace operation create the label if it doesn't exist? I noticed that the annotation isn't set in the pod spec, how does that affect the json patch and downwards api?

Reply:
I just checked and its actually not supposed to, but for some reason it's working. We can do a check first though